src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/net/NetworkRequestQueue.java

```diff
@@ -0,0 +1,51 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.net;
+
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+/**
+ * Class that represents a custom bounded and concurrent queue.
+ */
+public class NetworkRequestQueue<T> {
+
+  private static final int DEFAULT_MAX_Q_SIZE = 200;
+
+  private final BlockingQueue<T> messageQueue;
+
+  public NetworkRequestQueue() {
+    this(DEFAULT_MAX_Q_SIZE);
+  }
+
+  public NetworkRequestQueue(final int queueSize) {
+    this.messageQueue = new ArrayBlockingQueue<>(queueSize);
+  }
+
+  /**
+   * Adds a message to the queue if not full.
+   *
+   * @param message The message that needs to be enqueued..
+   * @return true if successfully enqueued, false otherwise.
+   */
+  public boolean offer(final T message) {
+    // happens-before (java.util.concurrent collection)
```

Comment:
Do we want to remove all the `happens-before` comments ?

Reply:
I will get rid of it after the review.